gunrock/app/vn/vn_app.cu

```diff
@@ -0,0 +1,313 @@
+    {
+        problem.Reset(srcs, num_srcs, target);
+        enactor.Reset(srcs, num_srcs, target);
+
+        cpu_timer.Start();
+        enactor.Enact(srcs);
+        cpu_timer.Stop();
+
+        total_time += cpu_timer.ElapsedMillis();
+        problem.Extract(distances, preds == NULL ? NULL : preds);
+    }
+
+    enactor.Release(target);
+    problem.Release(target);
+    srcs_vector.clear();
+    return total_time;
+}
+
+
+//  * @brief Simple interface take in graph as CSR format
+//  * @param[in]  num_nodes   Number of veritces in the input graph
+//  * @param[in]  num_edges   Number of edges in the input graph
+//  * @param[in]  row_offsets CSR-formatted graph input row offsets
+//  * @param[in]  col_indices CSR-formatted graph input column indices
+//  * @param[in]  edge_values CSR-formatted graph input edge weights
+//  * @param[in]  num_runs    Number of runs to perform vn
+//  * @param[in]  sources     Sources to begin traverse, one for each run
+//  * @param[in]  mark_preds  Whether to output predecessor info
+//  * @param[out] distances   Return shortest distance to source per vertex
+//  * @param[out] preds       Return predecessors of each vertex
+//  * \return     double      Return accumulated elapsed times for all runs
+ 
+
+template <
+    typename VertexT = int,
+    typename SizeT   = int,
+    typename GValueT = unsigned int,
+    typename vnValueT = GValueT>
+double vn(
+    const SizeT        num_nodes,
+    const SizeT        num_edges,
+    const SizeT       *row_offsets,
+    const VertexT     *col_indices,
+    const GValueT     *edge_values,
+    const int          num_runs,
+          VertexT     *sources,
+    const bool         mark_pred,
+          vnValueT    *distances,
+          VertexT     *preds = NULL)
+{
+    typedef typename gunrock::app::TestGraph<VertexT, SizeT, GValueT,
+        gunrock::graph::HAS_EDGE_VALUES | gunrock::graph::HAS_CSR>
+        GraphT;
+    typedef typename GraphT::CsrT CsrT;
+
+    // Setup parameters
+    gunrock::util::Parameters parameters("vn");
+    gunrock::graphio::UseParameters(parameters);
+    gunrock::app::vn::UseParameters(parameters);
+    gunrock::app::UseParameters_test(parameters);
+    parameters.Parse_CommandLine(0, NULL);
+    parameters.Set("graph-type", "by-pass");
+    parameters.Set("mark-pred", mark_pred);
+    parameters.Set("num-runs", num_runs);
+    std::vector<VertexT> srcs;
+    for (int i = 0; i < num_runs; i ++)
+        srcs.push_back(sources[i]);
+    
+    parameters.Set("srcs", srcs);
+
+    bool quiet = parameters.Get<bool>("quiet");
+    GraphT graph;
+    // Assign pointers into gunrock graph format
+    graph.CsrT::Allocate(num_nodes, num_edges, gunrock::util::HOST);
+    graph.CsrT::row_offsets   .SetPointer(row_offsets, gunrock::util::HOST);
+    graph.CsrT::column_indices.SetPointer(col_indices, gunrock::util::HOST);
+    graph.CsrT::edge_values   .SetPointer(edge_values, gunrock::util::HOST);
```

Comment:
293-295: Need the size parameter in the center, right now 1 is being passed as size; use num_nodes, num_edges, etc as you did in the hella_app.cu.

Reply:
https://github.com/gunrock/gunrock/pull/378#discussion_r214112605

Reply:
Patched the SSSP issue here: https://github.com/gunrock/gunrock/pull/385

Reply:
Updated `vn_app.cu` in `bkj/vn`